R/zzz.R

```diff
@@ -1,5 +1,27 @@
 .lockbox_env <- new.env()
 
+.get_safely_from_envir <- function(target, default = NULL) {
+  res <- tryCatch(
+    get(target, envir = .lockbox_env)
+  , error = function(e) {default})
+}
+
+set_session_id <- function() {
+  if (is.null(.get_safely_from_envir("session_id"))) {
+    if (requireNamespace("uuid", quietly = TRUE)) {
+      assign("session_id", uuid::UUIDgenerate(), envir = .lockbox_env)
```

Comment:
I want to be sure we don't risk getting a new session_id in the middle of a session.  I'm thinking of adding an early escape `if (!is.null(.lockbox_env$session_id)) {return()}` like you've done elsewhere to help be sure of that, then for the rest of the code we aren't stuck a level of nesting deep.  This also gets the symlink cleanup inside the protection which is where it belongs.